packages/ember-data/lib/system/store.js

```diff
@@ -1042,21 +1042,22 @@ Store = Service.extend({
 
     @method query
     @param {String} modelName
-    @param {any} query an opaque query to be used by the adapter
+    @param {Hash} options, query an opaque query of the form {query: { any }} to be used by the adapter
     @return {Promise} promise
   */
-  query: function(modelName, query) {
+  query: function(modelName, options) {
+    Ember.assert("You need to pass a query hash in the options hash of the store's query method", options && options.query);
```

Comment:
That works for me. I believe it should log a deprecate if it detects `options` does not have a `query` property.